packages/app-store/salesforce/components/components/WriteToObjectSettings.tsx

```diff
@@ -98,77 +159,213 @@ const WriteToObjectSettings = ({
             <div className="flex-1">{t("field_type")}</div>
             <div className="flex-1">{t("value")}</div>
             <div className="flex-1">{t("when_to_write")}</div>
-            <div className="w-10" />
+            <div className="w-20" />
           </div>
           <Section.SubSectionNested>
-            {Object.keys(writeToObjectData).map((key) => (
-              <div className="flex items-center gap-2" key={key}>
-                <div className="flex-1">
-                  <InputField value={key} readOnly size="sm" className="w-full" />
-                </div>
-                <div className="flex-1">
-                  <Select
-                    size="sm"
-                    className="w-full"
-                    value={fieldTypeOptions.find(
-                      (option) => option.value === writeToObjectData[key].fieldType
+            {Object.keys(writeToObjectData).map((key) => {
+              const isEditing = editingRows[key];
+              const editData = editingData[key];
+
+              return (
+                <div className="flex items-center gap-2" key={key}>
+                  <div className="flex-1">
+                    {isEditing ? (
+                      <InputField
+                        value={editData?.field || key}
+                        onChange={(e) =>
+                          setEditingData((prev) => ({
+                            ...prev,
+                            [key]: { ...editData, field: e.target.value },
+                          }))
+                        }
+                        size="sm"
+                        className="w-full"
+                      />
+                    ) : (
+                      <InputField value={key} readOnly size="sm" className="w-full" />
```

Comment:
I see significant duplication here b/w editing and readonly mode. Could we not achieve the same behaviour by simply adding _readOnly_ attribute conditionally